components/brave_stats/browser/brave_stats_updater_util.cc

```diff
@@ -131,4 +133,36 @@ uint8_t UsageBitfieldFromTimestamp(const base::Time& last_usage_time,
   return result;
 }
 
+bool GetLatestBrowserVersionFromPingResponse(
+    const std::string& json,
+    std::string* latest_version_result) {
+  base::JSONReader::ValueWithError parsed_value =
+      base::JSONReader::ReadAndReturnValueWithError(
+          json, base::JSONParserOptions::JSON_PARSE_RFC);
+  if (!parsed_value.value) {
+    VLOG(1)
+        << "Latest browser version retrieval: Bad usage ping response JSON. "
+        << "Error:" << parsed_value.error_message;
+    return false;
+  }
+
+  const base::DictionaryValue* resp_dict;
+  if (!parsed_value.value->GetAsDictionary(&resp_dict)) {
```

Comment:
you can skip this step, just check that `parsed_value.value->is_dict()` in the `if` above and then do FindStringKey